src/main/scala/ch/epfl/bluebrain/nexus/storage/Storages.scala

```diff
@@ -212,24 +217,26 @@ object Storages {
           .runWith(Sink.last)
           .to[F]
 
-      if (!Files.exists(absSourcePath))
-        F.pure(Left(PathNotFound(name, sourceRelativePath)))
-      else if (!absSourcePath.descendantOf(bucketPath) || absSourcePath.descendantOf(bucketProtectedPath))
-        F.pure(Left(PathNotFound(name, sourceRelativePath)))
-      else if (!absDestPath.descendantOf(bucketProtectedPath))
-        F.raiseError(PathInvalid(name, destRelativePath))
-      else if (Files.exists(absDestPath))
-        F.pure(Left(PathAlreadyExists(name, destRelativePath)))
-      else if (Files.isSymbolicLink(absSourcePath) || containsHardLink(absSourcePath))
-        F.pure(Left(PathContainsLinks(name, sourceRelativePath)))
-      else if (Files.isRegularFile(absSourcePath))
-        computeSizeAndMove(isDir = false)
-      else if (Files.isDirectory(absSourcePath))
-        dirContainsLink(absSourcePath).flatMap {
-          case true  => F.pure(Left(PathContainsLinks(name, sourceRelativePath)))
-          case false => computeSizeAndMove(isDir = true)
-        }
-      else F.pure(Left(PathNotFound(name, sourceRelativePath)))
+      fixPermissions(absSourcePath).flatMap { fixPermsResult =>
```

Comment:
if fix permissions fails, why continue with the rest of the checks?

(I don't understand the need for the return type `F[Either[PermissionsFixingFailed, Unit]]`...)

Reply:
because we want to return a meaningful error to the client:
- PathNotFound
- PathContainsLinks
...
are more meaningful than an error on permissions.

in the case all the other errors are not thrown and we would proceed to moving the file, then the permission fixer error will be thrown

Reply:
makes sense.